### PR TITLE
[SPARK-17150][SQL] Support SQL generation for inline tables

### PR DESCRIPTION
**Pull request by petermaxlee**
## What changes were proposed in this pull request?

This patch adds support for SQL generation for inline tables. With this, it would be possible to create a view that depends on inline tables.
## How was this patch tested?

Added a test case in LogicalPlanToSQLSuite.

### CODE REVIEW

**Comment by petermaxlee** [comment]
cc @cloud-fan and @hvanhovell

**Comment by rxin** [comment]
I suspect array and struct literals will fail, looking at what Literal.sql does. That said, it's an existing problem and we can fix that later.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LocalRelation.scala [hunk 1]
```diff
@@ -75,4 +76,16 @@ case class LocalRelation(output: Seq[Attribute], data: Seq[InternalRow] = Nil)
 
   override lazy val statistics =
     Statistics(sizeInBytes = output.map(_.dataType.defaultSize).sum * data.length)
+
+  def toSQL(inlineTableName: String): String = {
+    require(data.nonEmpty)
+    val types = schema.fields.map(_.dataType)
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LocalRelation.scala]: nit: `output.map(_.dataType)`

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LocalRelation.scala [hunk 2]
```diff
@@ -75,4 +76,16 @@ case class LocalRelation(output: Seq[Attribute], data: Seq[InternalRow] = Nil)
...
+    val rows = data.map { row =>
+      val cells = row.toSeq(types).zip(types).map { case (v, tpe) => Literal(v, tpe).sql }
+      cells.mkString("(", ", ", ")")
+    }
+    "VALUES " + rows.mkString(", ") +
+      " AS " + inlineTableName +
+      schema.fields.map(_.name).mkString("(", ", ", ")")
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LocalRelation.scala]: nit: `output.map(_.name)`

**Comment by cloud-fan** [comment]
retest this please

**Comment by cloud-fan** [comment]
thanks, merging to master/2.0

**Pull request closed without merging**